prow/github/client.go

```diff
@@ -84,6 +84,10 @@ var (
 
 const (
 	acceptNone = ""
+	// Abort requests that don't return in 10 mins. Longest graphql calls can
+	// take up to 2 minutes. This limit should ensure all successful calls return
+	// but will prevent an indefinite stall if GitHub never responds.
+	maxRequestTime = 10 * time.Minute
```

Comment:
is 10 minutes really reasonable? that seems excessively long for an HTTP request

Reply:
OK. 2 to 3 times our already large upper bound seems more reasonable than 5x. Hanging for 10 minutes in the hopes that we might actually get a response before then seems excessive.

Reply:
We could also set a separate limit for v3 and v4 API. v3 should be faster and more reliable so we could set a shorter time limit for the v3 `http.Client` if that is desired, but I don't know what the maximum successful response time is for v3 API requests.

Reply:
let's get this deployed tomorrow and follow-up with improving it.